### PR TITLE
Add code to `setup.py` to on-the-fly convert the contents of the `README.md` to the ReStructedText required by PyPI

### PR DESCRIPTION
**Pull request by alexander255**
Thanks to this `./setup.py register` now "just works" and will upload the current `README.md` and all information in `setup.py` to PyPI.

Also I fixed some minor issues in the README.

### CODE REVIEW

**Comment by alexander255** [comment]
It would also be great if at least one other person with access to this repository (particularily @whereswaldon) could register an account with PyPI and send my their account name so that I can add them to https://pypi.python.org/pypi/ipfsapi .

**Comment by whereswaldon** [comment]
You can add me. My username is the same as github.

**Comment by alexander255** [comment]
Since this commit finalizes the breaking changes we have had streaming in, in the last month or two I've also added a section to the `README` summarizing all breaking changes done.

**Comment by alexander255** [comment]
Also fixed crash in Python 3.x if `pandoc` is not installed.

File: README.md [hunk 1]
```diff
@@ -4,21 +4,23 @@
...
+[![](https://img.shields.io/pypi/v/ipfsapi.svg?style=flat-square)](https://pypi.python.org/pypi/ipfsapi)
 [![](https://img.shields.io/travis/ipfs/py-ipfs-api.svg?style=flat-square)](https://travis-ci.org/ipfs/py-ipfs-api)
 
 > A python client library for the IPFS API
 
 Check out [the client API reference](https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/) for the full command reference.
 
 **Note:** This library constantly has to change to stay compatible with the IPFS HTTP API.
-Currently, this library is tested against [go-ipfs v0.4.3rc3](https://github.com/ipfs/go-ipfs/releases/tag/v0.4.3-rc3).
+Currently, this library is tested against [go-ipfs v0.4.3rc4](https://github.com/ipfs/go-ipfs/releases/tag/v0.4.3-rc4).
```
Comment (whereswaldon) [inline README.md]: Nice catch. Need to remember to bump that whenever we update the Travis file.

File: README.md [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
 You may experience compatibility issues when attempting to use it with other versions of go-ipfs.
 
 ## Table of Contents
 
 - [Install](#install)
 - [Usage](#usage)
+- [Documentation](#documentation)
```
Comment (whereswaldon) [inline README.md]: Have you reviewed the [Standard README](https://github.com/RichardLitt/standard-readme/blob/master/spec.md) stuff that the IPFS team put together? This follows that spec, but I was curious whether that was deliberate or coincidence.

File: README.md [hunk 3]
```diff
@@ -113,7 +115,22 @@ Documentation (currently mostly API documentation unfortunately) is available on
...
-The [`ipfs` command-line Client documentation](https://ipfs.io/docs/commands/) may also be useful in some cases.
+The `ipfs` [command-line Client documentation](https://ipfs.io/docs/commands/) may also be useful in some cases.
+
+### Important changes from `ipfsApi 0.2.x`
+
+ * The Python package has been renamed from `ipfsApi` to `ipfsapi`
+ * The PIP module has been renamed from `ipfs-api` to `ipfsapi` (please update your requirement files)
+ * A lot of changes in the internal code
+    - Commands have been completely removed
+    - Usage of `requests` or other libraries is considered an implementation detail from now on
```
Comment (whereswaldon) [inline README.md]: Was it not considered an implementation detail before? I'm not sure that I understand what you mean.

File: README.md [hunk 4]
```diff
@@ -113,7 +115,22 @@ Documentation (currently mostly API documentation unfortunately) is available on
...
+ * Most parts of the library (except for `Client()`) are now considered internal and may therefor break at any time
```
Comment (whereswaldon) [inline README.md]: Sorry if this seems pedantic, but therefore needs an "e" at the end.

File: setup.py [hunk 5]
```diff
@@ -10,6 +15,44 @@
     exec(file.read())
 
 
+def get_long_description():
```
Comment (whereswaldon) [inline setup.py]: It took me a while to grok what this is doing. Nicely done. Looks very thorough. Where have you been able to test it (on which operating systems, I mean)?

File: README.md (hunk 2 above)
Comment (alexander255) [inline README.md]: I've quickly skimmed it once (when we got the new badges), but I didn't remember any of it. The compliance is therefor purely coincidental :wink:

File: README.md (hunk 4 above)
Comment (alexander255) [inline README.md]: std::typo, thanks for pointing it out!

File: setup.py (hunk 5 above)
Comment (alexander255) [inline setup.py]: Just my Linux/Debian system, I hope it works everywhere :grin:

File: README.md (hunk 3 above)
Comment (whereswaldon) [inline README.md]: Okay, thanks for clarifying.

File: setup.py (hunk 5 above)
Comment (whereswaldon) [inline setup.py]: Uh-oh. I'm also running Linux (only Ubuntu), so our cross-platform testing isn't terribly thorough.

**Pull request merged**